numba/cgutils.py

```diff
@@ -47,6 +48,117 @@ def make_bytearray(buf):
     n = len(b)
     return ir.Constant(ir.ArrayType(ir.IntType(8), n), b)
 
+_struct_proxy_cache = {}
+
+
+def create_struct_proxy(fe_type):
+    """
+    Returns a specialized StructProxy subclass for the given fe_type.
+    """
+    res = _struct_proxy_cache.get(fe_type)
+    if res is None:
+        clsname = StructProxy.__name__ + '_' + str(fe_type)
+        bases = (StructProxy,)
+        clsmembers = dict(_fe_type=fe_type)
+        res = type(clsname, bases, clsmembers)
+        _struct_proxy_cache[fe_type] = res
+    return res
+
+
+class StructProxy(object):
+    """
+    Creates a `Structure` like interface that is constructed with information
```

Comment:
What's the difference with a regular Structure? You mean it takes LLVM types instead of Numba types in the fields description? Also, IIUC there's a lot of duplicated code between those two classes :-)